packages/react-dom/src/__tests__/ReactTestUtils-test.js

```diff
@@ -573,33 +520,45 @@ describe('ReactTestUtils', () => {
       }
     }
 
-    const container = document.createElement('div');
-    const root = ReactDOMClient.createRoot(container);
-    await act(() => {
-      root.render(<Component />);
-    });
-
+    ReactTestUtils.renderIntoDocument(<Component />);
     expect(mockArgs.length).toEqual(0);
   });
-  it('should find rendered component with type in document', async () => {
+
+  // @gate !disableDOMTestUtils
+  it('should find rendered component with type in document', () => {
     class MyComponent extends React.Component {
       render() {
         return true;
       }
     }
 
-    const container = document.createElement('div');
-    const root = ReactDOMClient.createRoot(container);
-    let instance;
-    await act(() => {
-      root.render(<MyComponent ref={current => (instance = current)} />);
-    });
-
+    const instance = ReactTestUtils.renderIntoDocument(<MyComponent />);
     const renderedComponentType = ReactTestUtils.findRenderedComponentWithType(
       instance,
       MyComponent,
     );
 
     expect(renderedComponentType).toBe(instance);
   });
+
+  // @gate disableDOMTestUtils
+  it('throws on every removed function with a special message', async () => {
+    expect(ReactTestUtils.isDOMComponent).toThrowError(
+      '`isDOMComponent` was removed from `react-dom/test-utils`. ' +
+        'For testing React, we recommend React Testing Library instead: https://testing-library.com/docs/react-testing-library/intro.',
```

Comment:
Similar to the react-test-renderer warning, I don't think we should hard code the recommendations into the error message, instead something like:

> `isDOMComponent` was removed from `react-dom/test-utils`. See https://react.dev/warnings/react-test-renderer for more info.

Reply:
Ditto for the other warning messages added here.

Reply:
And we can update that warning page with info on which methods were removed, and why we removed them.